src/backend/utils/adt/numeric.c

```diff
@@ -5906,8 +5906,11 @@ numeric_poly_sum(PG_FUNCTION_ARGS)
 #endif
 }
 
+/*
+ *	Final function for BIGINT datatype for TSQL Aggregates (SUM,AVG)
+ */
 Datum
-bigint_poly_sum(PG_FUNCTION_ARGS)
+bigint_utility(FunctionCallInfo fcinfo, int tsqlAggType)
```

Comment:
1. The current function name is a little bit general. Can we use a more specific name, e.g., `bigint_poly_aggr` or `bigint_aggr_final`?
2. Instead of implementing our own logic, can we leverage `numeric_poly_sum` and `numeric_poly_avg` in the function body? We may call the existing function and then check overflow.

Reply:
Regarding point 2:-
1. In T-Sql, The average aggregates  doesn't calculate the average of values first and  then checks for bounds but rather sums up the values and checks if its out of bound , if not then it proceeds in calculating the average, So to match this behaviour we have introduced our own logic
2.  If we use numeric_poly_sum / numeric_poly_average we would be get the value/result back in numeric which will again need additional processing, So I didn't use them in the common utility function

Reply:
Fixed 1. in latest commit